operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/Condition.java

```diff
@@ -10,12 +11,10 @@
    * {@link io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource} based on the
    * observed cluster state.
    *
+   * @param dependentResource for which the condition applies to
    * @param primary the primary resource being considered
-   * @param secondary the secondary resource associated with the specified primary resource or
-   *        {@code null} if no such secondary resource exists (for example because it's been
-   *        deleted)
    * @param context the current reconciliation {@link Context}
    * @return {@code true} if the condition holds, {@code false} otherwise
    */
-  boolean isMet(P primary, R secondary, Context<P> context);
+  boolean isMet(DependentResource<R, P> dependentResource, P primary, Context<P> context);
```

Comment:
Maybe keep the order that was there before (i.e. primary as first param)?

Reply:
Hmm, don't, have a strong opinion on that. Changed this because the condition applies primarily to a dependent resource, so that's why it is on first place (kinda leading or priority)